### PR TITLE
Added logout command.

### PR DESCRIPTION
**Pull request by mkreibe**
This pull request is to add the logout as described in [Issue 660](https://github.com/concourse/concourse/issues/660). It is currently Tracker issue [130789427](https://www.pivotaltracker.com/n/projects/1059262/stories/130789427).

The `rc/targets.go` file was updated to have just one command to get the `.flyrc` file too.

### CODE REVIEW

File: rc/targets.go [hunk 1]
```diff
@@ -40,6 +40,24 @@ type targetDetailsYAML struct {
 	Targets map[TargetName]TargetProps
 }
 
+func getFileName() (flyrc string) {
```
Comment (vito) [inline rc/targets.go]: this should be called something like `flyrcPath` instead

also named return values are discouraged; could just be `return filepath.Join(...)` instead anyway

File: commands/logout.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	"errors"
+
+	"github.com/concourse/fly/rc"
+)
+
+type LogoutCommand struct {
+	All bool `short:"a" long:"all" description:"Logout of all targets"`
+}
+
+func (command *LogoutCommand) Execute(args []string) (err error) {
```
Comment (vito) [inline commands/logout.go]: please avoid using named return values

File: commands/logout.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		}
+
+		for targetName, _ := range flyYAML.Targets {
+			if err = rc.DeleteTarget(targetName); err != nil {
+				return err
+			}
+		}
+
+	} else {
+		err = errors.New("Must specify either a target (--target/-t) or the all flag (--all/-a)")
```
Comment (vito) [inline commands/logout.go]: error messages should be lowercase; how about just `must specify --target or --all`?

File: commands/fly.go [hunk 4]
```diff
@@ -10,8 +10,9 @@ type FlyCommand struct {
 
 	Version func() `short:"v" long:"version" description:"Print the version of Fly and exit"`
 
-	Login LoginCommand `command:"login" alias:"l" description:"Authenticate with the target"`
-	Sync  SyncCommand  `command:"sync"  alias:"s" description:"Download and replace the current fly from the target"`
+	Login  LoginCommand  `command:"login" alias:"l" description:"Authenticate with the target"`
+	Logout LogoutCommand `command:"logout" alias:"lo" description:"Release authentication with the target"`
```
Comment (vito) [inline commands/fly.go]: conventionally single-word commands have a single-character alias. how about just `o`?

File: rc/targets.go (hunk 1 above)
Comment (mkreibe) [inline rc/targets.go]: Done

File: commands/logout.go (hunk 2 above)
Comment (mkreibe) [inline commands/logout.go]: Done

File: commands/logout.go (hunk 3 above)
Comment (mkreibe) [inline commands/logout.go]: Done

File: commands/fly.go (hunk 4 above)
Comment (mkreibe) [inline commands/fly.go]: Done

**Comment by mkreibe** [comment]
I agree with the tests... I haven't found *any* for this project... maybe I'm looking in the wrong location?

**Comment by vito** [comment]
Check under `integration/` - maybe base them on `login` (the simpler cases)?

**Comment by mkreibe** [comment]
great, I'll get on it.

**Pull request closed without merging**